eland/operations.py

```diff
@@ -626,14 +669,68 @@ def _unpack_metric_aggs(
                         else:
                             agg_value = field.np_dtype.type(agg_value)
 
-                values.append(agg_value)
+                if not percentile_values:
+                    values.append(agg_value)
 
             # If numeric_only is True and We only have a NaN type field then we check for empty.
             if values:
                 results[field.column] = values if len(values) > 1 else values[0]
+            # This only runs when df.quantile() or series.quantile() is called
+            if percentile_values and not is_dataframe_agg:
+                results[f"{field.column}"] = percentile_values
 
         return results
 
+    def quantile(
+        self,
+        query_compiler: "QueryCompiler",
+        pd_aggs: List[str],
+        quantiles: Union[int, float, List[int], List[float]],
+        is_dataframe: bool = True,
+        numeric_only: Optional[bool] = True,
+    ) -> Union[pd.DataFrame, pd.Series]:
+        # To verify if quantile range falls between 0 to 1
+        def quantile_to_percentile(quantile: Any) -> float:
+            if isinstance(quantile, (int, float)):
+                quantile = float(quantile)
+                if quantile > 1 or quantile < 0:
+                    raise ValueError(
+                        f"quantile should be in range of 0 and 1, given {quantile}"
+                    )
+            else:
+                raise TypeError("quantile should be of type int or float")
+            # quantile * 100 = percentile
+            # return float(...) because min(1.0) gives 1
+            return float(min(100, max(0, quantile * 100)))
+
+        percentiles = [
+            quantile_to_percentile(x)
+            for x in (
+                (quantiles,) if not isinstance(quantiles, (list, tuple)) else quantiles
+            )
+        ]
+
+        result = self._metric_aggs(
+            query_compiler,
+            pd_aggs=pd_aggs,
+            percentiles=percentiles,
+            is_dataframe_agg=False,
+            numeric_only=numeric_only,
+        )
+
+        df = pd.DataFrame(
+            result,
+            index=[i / 100 for i in percentiles],
+            columns=result.keys(),
+            dtype=(np.float64 if numeric_only else None),
+        )
+
+        # Display Output same as pandas does
+        if isinstance(quantiles, float):
+            return df.squeeze() if is_dataframe else df.squeeze()
```

Comment:
This `... if is_dataframe else ...` calls `df.squeeze()` either way? Should we be treating these two scenarios differently? Can you point out where these two scenarios are being tested so I can verify it's the same as pandas too.

Reply:
Ultimate Noob Mistake Sorry

So, If a single quantile is given. then
for dataframe => output here is a dataframe ,we convert into series (because pandas output is series)
```python
>>> print(pd_flights.quantile(q=0.2, numeric_only=True))
AvgTicketPrice    360.798081
FlightDelayMin      0.000000
dayOfWeek           1.000000
Name: 0.2, dtype: float64

>>>print(ed_flights.quantile(q=0.2, numeric_only=True))
df
     AvgTicketPrice  FlightDelayMin  dayOfWeek
0.2      361.040768             0.0        1.0
df.squeeze()
AvgTicketPrice    361.040768
FlightDelayMin      0.000000
dayOfWeek           1.000000
Name: 0.2, dtype: float64
```



For series we get a single column and single row and pandas squeezes its output
```python
# Eland Example :
>>> print(ed_flights["AvgTicketPrice"].quantile(.2))
df
     AvgTicketPrice
0.2      361.040768
df.squeeze()
361.0407676152134

>>> print(pd_flights["AvgTicketPrice"].quantile(.2))
360.7980812247401

```